File: 0392-is-subsequence/0392-is-subsequence.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
        int i = 0, sSize = s.length();
        for(char c: t){
            if(s[i] == c) i++;
            if(i == sSize) return true;
        }
        if(i == sSize) return true;
        return false;
    }
};

File: 0392-is-subsequence/0392-is-subsequence.py
class Solution(object):
    def isSubsequence(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        i, size = 0, len(s)
        for c in t:
            if i < size and c == s[i]:
                i += 1
        
        if i < size:
            return False
        return True
        